File: backend/QuizService/src/main/java/com/quizarena/quiz/feature/quiz/model/QuizCategory.java
package com.quizarena.quiz.feature.quiz.model;

public enum QuizCategory {
    GENERAL_KNOWLEDGE("General Knowledge"),
    SCIENCE_AND_NATURE("Science & Nature"),
    HISTORY("History"),
    GEOGRAPHY("Geography"),
    SPORTS("Sports"),
    ENTERTAINMENT("Entertainment"),
    ART_AND_LITERATURE("Art & Literature"),
    TECHNOLOGY("Technology"),
    MATHEMATICS("Mathematics"),
    POP_CULTURE("Pop Culture");

    private final String displayName;

    QuizCategory(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}

File: frontend/src/app/features/quizzes/types/quiz-category.ts
export enum QuizCategory {
  GENERAL_KNOWLEDGE = 'GENERAL_KNOWLEDGE',
  SCIENCE_AND_NATURE = 'SCIENCE_AND_NATURE',
  HISTORY = 'HISTORY',
  GEOGRAPHY = 'GEOGRAPHY',
  SPORTS = 'SPORTS',
  ENTERTAINMENT = 'ENTERTAINMENT',
  ART_AND_LITERATURE = 'ART_AND_LITERATURE',
  TECHNOLOGY = 'TECHNOLOGY',
  MATHEMATICS = 'MATHEMATICS',
  POP_CULTURE = 'POP_CULTURE',
}

export const QUIZ_CATEGORY_LABELS: Record<QuizCategory, string> = {
  [QuizCategory.GENERAL_KNOWLEDGE]: 'General Knowledge',
  [QuizCategory.SCIENCE_AND_NATURE]: 'Science & Nature',
  [QuizCategory.HISTORY]: 'History',
  [QuizCategory.GEOGRAPHY]: 'Geography',
  [QuizCategory.SPORTS]: 'Sports',
  [QuizCategory.ENTERTAINMENT]: 'Entertainment',
  [QuizCategory.ART_AND_LITERATURE]: 'Art & Literature',
  [QuizCategory.TECHNOLOGY]: 'Technology',
  [QuizCategory.MATHEMATICS]: 'Mathematics',
  [QuizCategory.POP_CULTURE]: 'Pop Culture',
};
